core/types/calls.cc

```diff
@@ -415,6 +415,7 @@ MethodRef guessOverload(const GlobalState &gs, ClassOrModuleRef inClass, MethodR
 
         // If keyword args are present, interpret them as an untyped hash
         if (numPosArgs < args.size()) {
+            // TODO(neil): this should actually blame to an error recovery magic symbol
```

Comment:
Why would this blame to an error recovery magic symbol? If I'm reading this right, there's no error we're recovering from here.
 
If anything, I think this should blame to something that would indicate we need better support for ruby 3 keyword arg handling.

But it's also possible that we don't want to track anything here: this doesn't _really_ count as a usage of untyped—rather, it's Sorbet that uses `untyped` as a convenient way to guess an overload. Once that overload is guessed, it will be used for type checking later on, where if any of the keyword args are untyped, they will be checked and have their untyped usages reported.

Reply:
>  this doesn't really count as a usage of untyped—rather, it's Sorbet that uses untyped as a convenient way to guess an overload.

Yup, this is the mental model I had too, that this is a transient untyped object, and that it will go away once the overload is checked (and that's why I called  the `hashOfUntyped` overload without a blame).

I wrote error recovery symbol just as a default "we don't care about this". I'll drop the comment and leave it as-is (and we create a new ignore symbol if we really care about removing all `untypedUntracked` uses).

Reply:
ah, that makes sense!